### ISSUE TITLE
Reauthorize Trakt

### ISSUE THREAD

**Issue by brentonv** [jurialmunkey/plugin.video.themoviedb.helper#191]
Hi there. I had this strange issue today where no TMdBhelper widgets were loading. The Kodi status bar said TMdBhelper errors so after checking the log it showed it was simply Trakt authentication failure.

So in saying that would there be a way to make the Authenticate Trakt window automatically appear if the add-on can't authenticate. Just to make it simpler.

Thanks

**Comment by drinfernoo** [comment]
I'm not sure if every menu does it, but I definitely have had certain widgets load and prompt for authorization.

**Comment by brentonv** [comment]
I thought so too. I did exit Kodi a few times to see if it would initiate but didn't so went into settings. Maybe just a glitch. I'm on Xbox which behaves unique at times :)

**Comment by drinfernoo** [comment]
@brentonv You could just authorize Trakt and they'd work 😅

**Comment by brentonv** [comment]
What? Yes that would work but I'm explaining that Trakt was previously authorised and displaying Trakt widgets but yesterday no widgets loaded and TMdBhelper was showing errors. I wasn't aware it was Trakt authorisation until I viewed the Kodi log. Figured I would report it incase it's a recent bug.

Like you said the Authentication window should have displayed but it didn't. And it didn't display from the settings menu either the first three times I tried to open it. After closing Kodi again and waiting for the errors to pass it did open. Odd behaviour :confused:

**Comment by jurialmunkey** [comment]
Can you provide the log please?

If there was an error notification then that indicates a raised error in the code which needs to be fixed.

**Comment by brentonv** [comment]
Sorry no I didn't keep the log. And I've checked my Android Kodi there's no issues there. But I just checked  my add-on update history and think I've found the issue.

My system does a backup on the first day of the month and set to backup on next launch if missed. Looks like as it was loading yesterday it was backing up while at the same time updating quite a few add-ons which is bad design by the backup.

So as I expected TMdBHelper updated yesterday on launch. While at the same time backing up. And trying to display widgets. And it's Xbox lol.

Everything went back to normal after reauthorising Trakt. And also I deleted the settings.xml and yes the auth window pops up as you said it should.

So just a glitch on my end. Thanks guys
